src/app/components/dynamic-list/DynamicList.jsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import DynamicListItem from "./DynamicListItem";
+import PropTypes from "prop-types";
+import Input from "../Input";
+
+const propTypes = {
+    title: PropTypes.string.isRequired,
+    listHeightClass: PropTypes.string,
+    filterPlaceholder: PropTypes.string,
+    noResultsText: PropTypes.string,
+    handleSearchInput: PropTypes.func,
+    listItems: PropTypes.arrayOf(PropTypes.shape(DynamicListItem.propTypes)).isRequired,
+};
+
+const DynamicList = props => {
+    const searchID = `${props.title.replace(/\s+/g, "-").toLowerCase()}-search-content-types`;
+    let list;
+    if (props.listItems.length > 0) {
+        list = (
+            <ul className={`dynamic-list__container ${props.listHeightClass}`}>
+                {props.listItems.map((row, index) => {
+                    return <DynamicListItem key={`dynamic-list-row-${index}`} {...row} />;
+                })}
+            </ul>
+        );
+    } else {
+        list = <div role="alert">{props.noResultsText}</div>;
+    }
+    return (
+        <div>
+            <span className="dynamic-list__title">{props.title}</span>
```

Comment:
If this is a title, should we use a suitable heading element instead of `span`? May require additional CSS to override any global heading tag styles, but it feels like it would be more semantically accurate